lib/web_ui/lib/src/engine/compositor/surface.dart

```diff
@@ -119,51 +121,75 @@ class Surface {
       ..position = 'absolute'
       ..width = '${logicalSize.width.ceil()}px'
       ..height = '${logicalSize.height.ceil()}px';
-    final int glContext = canvasKit.callMethod('GetWebGLContext', <dynamic>[
-      htmlCanvas,
-      // Default to no anti-aliasing. Paint commands can be explicitly
-      // anti-aliased by setting their `Paint` object's `antialias` property.
-      js.JsObject.jsify({'antialias': 0}),
-    ]);
-    _grContext =
-        canvasKit.callMethod('MakeGrContext', <dynamic>[glContext]);
-
-    if (_grContext == null) {
-      throw CanvasKitError('Could not create a graphics context.');
-    }
 
-    // Set the cache byte limit for this grContext, if not specified it will use
-    // CanvasKit's default.
-    _syncCacheBytes();
+    htmlElement = htmlCanvas;
+    if (canvasKitForceCpuOnly) {
+      return _makeSoftwareCanvasSurface(htmlCanvas);
+    } else {
+      // Try WebGL first.
+      final int glContext = canvasKit.callMethod('GetWebGLContext', <dynamic>[
+        htmlCanvas,
+        // Default to no anti-aliasing. Paint commands can be explicitly
+        // anti-aliased by setting their `Paint` object's `antialias` property.
+        js.JsObject.jsify({'antialias': 0}),
+      ]);
+
+      if (glContext == 0) {
+        return _makeSoftwareCanvasSurface(htmlCanvas);
+      }
+
+      _grContext =
+          canvasKit.callMethod('MakeGrContext', <dynamic>[glContext]);
+
+      if (_grContext == null) {
```

Comment:
If we were able to create a GL context but failed to create a graphics context then something is seriously wrong. I think we should throw here

Reply:
Done.